cdk_framework/EKS/lib/resource_constructs/general_constructs/general-sample-app-deployment-construct.ts

```diff
@@ -0,0 +1,85 @@
+import { PullModeSampleAppDeploymentConstruct } from '../specific_constructs/pull-mode-sample-app-construct';
+import { NamespaceConstruct } from '../specific_constructs/namespace-construct';
+import { SampleAppServiceConstruct } from '../specific_constructs/sample-app-service-construct';
+
+
+export class GeneralSampleAppDeploymentConstruct extends Construct {
+
+    constructor(scope: Construct, id: string, props: GeneralSampleAppDeploymentConstructProps){
+         super(scope, id);
+
+        new SampleAppServiceConstruct(this, 'sample-app-service-construct', {
+            cluster: props.cluster,
+            namespaceConstruct: props.namespaceConstruct,
+            sampleAppLabel: props.sampleAppLabel,
+            listenAddressPort: props.listenAddressPort
+        })
+
+        if (props.sampleAppMode === 'push'){
+            if (props.grpcServiceName == undefined) {
+                throw new Error('No GRPC Service name provided')
+            }
+            if (props.grpcPort == undefined) {
+                throw new Error('No GRPC port provided')
+            }
+            if (props.udpServiceName == undefined) {
+                throw new Error('No UDP Service name provided')
+            }
+            if (props.udpPort == undefined) {
+                throw new Error('No UDP port provided')
+            }
+            if (props.tcpServiceName == undefined) {
+                throw new Error('No TCP Service name provided')
+            }
+            if (props.httpPort == undefined) {
+                throw new Error('No HTTP port provided')
+            }
+            new PushModeSampleAppDeploymentConstruct(this, 'push-mode-sample-app-construct', {
+                cluster: props.cluster,
+                namespaceConstruct: props.namespaceConstruct,
+                sampleAppLabel: props.sampleAppLabel,
+                sampleAppImageURI: props.sampleAppImageURI,
+                grpcServiceName: props.grpcServiceName,
+                grpcPort: props.grpcPort,
+                udpServiceName: props.udpServiceName,
+                udpPort: props.udpPort,
+                tcpServiceName: props.tcpServiceName,
+                httpPort: props.httpPort,
+                listenAddressHost: props.listenAddressHost,
+                listenAddressPort: props.listenAddressPort,
+                region: props.region
+            })
+        }
+        else if (props.sampleAppMode === 'pull'){
+            new PullModeSampleAppDeploymentConstruct(this, 'pull-mode-sample-app-construct', {
+                cluster: props.cluster,
+                namespaceConstruct: props.namespaceConstruct,
+                sampleAppLabel: props.sampleAppLabel,
+                sampleAppImageURI: props.sampleAppImageURI,
+                listenAddressHost: props.listenAddressHost,
+                listenAddressPort: props.listenAddressPort,
+                region: props.region
```

Comment:
Can we use interfaces to expose these props and then do something like ` <push/pull>sampleAppProps = props as <push/pull>SampleAppProps`?

Reply:
We could export the props interface in the push and pull mode sample apps and instantiate an instance of that interface, but then the same code would be required to instantiate the appropriate interface and give it the required props. Maybe I'm misunderstanding the suggestion.